### ISSUE TITLE
"IGrabbable" output on "Can Be Grabbed" node

### ISSUE THREAD

**Issue by amplified1** [Yellow-Dog-Man/Resonite-Issues#1484]
### Is your feature request related to a problem? Please describe.

Right now the "Can Be Grabbed" node cannot distinguish between the 3(?) different components that make an object considered grabbable, there are also many useful applications to being able to use the other nodes associated with grabbables with an IGrabbable that has been dynamically accessed using the node.

### Describe the solution you'd like

Have the Can Be Grabbed node output the IGrabbable that makes it able to be grabbed.

### Describe alternatives you've considered

The node could also output just a string, enum, or otherwise of what type of grababble component it has found.

### Additional Context

The "Can Be Grabbed" component could be used in theory to make safe objects that interact with grabbable objects dynamically in the same way players can, but the lack of this functionality makes any attempt to do such a thing either much more complicated or much less safe, as it could potentially interfere with sliders and joints that were not intended to be moved

### Requesters

ShadowX, Amplified

**Comment by JackTheFoxOtter** [comment]
I'm not sure this would really fit into the node itself, but could make sense for a second one. A "get grabbable" node with a Slot and Grabber input as well, and the IGrabbable output. This would definitely make sense as a separate node, considering it needs to do the grab check and respect grabbable priority. (Also I've read & used the words "grabber" and "grabbable" so much now that my head is starting to spin a little :D)

**Comment by Frooxius** [comment]
Yes, this would probably be better as separate node to keep things clean. Have a "FindGrabbable" one.

However another thing is, the only thing "CanBeGrabbed" tells you is if a given player can grab given object with given grabber. This isn't intended to make you interact with those objects outside of that system.

**Comment by JackTheFoxOtter** [comment]
Understandable, but I'd say a "Find Grabbable" would be a logial extension to the functionality we already have now, and probably a relatively small addition.

**Comment by ProbablePrime** [comment]
For identifying the type of grabbable, the type nodes might work with this.

**Comment by Krzeszny** [comment]
Yup, I need this for my remover gun. I'd like to use a raycast to remove items in the same way as you grab them and choose Destroy in the context menu.

Currently, I get the explicit Object Root but lots of objects have multiple Object Roots and I end up removing a part of the object.

**Comment by Redd56** [comment]
dont need it rn but this would be amazing
